### ISSUE TITLE
Open interpreter not doing anything and not throwing any errors

### ISSUE THREAD

**Issue by davidmoshal** [KillianLucas/open-interpreter#393]
### Describe the bug

Installed open-interpreter per the docs.

running `interpreter` seems fine initially, but when asked to do anything at the prompt, it just writes a blank line!

### Reproduce

1. open terminal
2. type: `interpreter -y`
3. type any question
4. the result is always a blank line

### Expected behavior

all of these initially seem to work:

```
interpreter
interpreter -y
interpreter --local
interpreter --local -y
```

with then results in a prompt `>`
however no matter the question, I just get a blank line.

### Screenshots

<img width="458" alt="Screenshot 2023-09-15 at 4 56 11 PM" src="https://github.com/KillianLucas/open-interpreter/assets/650245/110b7957-6a5f-4a70-8975-54ec9b0e44db">

### Open Interpreter version

0.1.4

### Python version

3.11.3

### Operating System name and version

13.5.2

**Comment by rishabhjain** [comment]
Yep, same issue here.

**Comment by Tony72495** [comment]
I have a similar issue.  I'm getting successful responses from Chat-GPT, it lays out a plan of steps to accomplish a goal, and then....just nothing happens.  I tried it both in terminal and in a Python IDE, both result in no actual code being executed on the system.

**Comment by gagebt** [comment]
Same. Both in Colab and on Windows. 
Debug: Openai returns empty args for run_code. Looks like behaviour change on their side.

**Comment by seshakiran** [comment]
Same here

**Comment by ishaan-jaff** [comment]
what version of litellm are you on ?

**Comment by ishaan-jaff** [comment]
Can you run python3 -m poetry update litellm . This will ensure you're on the latest version of litellm. I want to make sure litellm is not causing anything to break for you

**Comment by seshakiran** [comment]
"Poetry could not find a pyproject.toml file in /Users/~mylocalsystem/Library/Application Support/Open Interpreter or its parents" -- this is the error I am getting after running the above. I installed poetry manually using pip

**Comment by cjbottaro** [comment]
When it gets stuck, you can ctrl-c a few times and kinda get back to a prompt.

```
  I apologize, but it seems that the required Python package "pyfiglet" is not installed in your environment. Would you like me to install it for you?

> yes
^CFailed to parse 'role'.
Exception in thread Thread-1391 (save_and_display_stream):
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/homebrew/lib/python3.11/site-packages/interpreter/code_interpreter.py", line 369, in save_and_display_stream
    raise KeyboardInterrupt
KeyboardInterrupt
^CFailed to parse 'content'.
^CFailed to parse 'role'.
> 
```

Also maybe that stack trace sheds some lights on where it gets stuck.

**Comment by davidmoshal** [comment]
> python3 -m poetry update litellm

`/opt/homebrew/opt/python@3.11/bin/python3.11: No module named poetry`

**Comment by davidmoshal** [comment]
Ok, I managed to get this to work, but running the project directly.

Note: my skill level with Python is not great (though excellent with the Node and JVM ecosystems), so most likely I'm doing it wrong, but it does work.

Steps:

1)  Installed Poetry per that project's docs
`curl -sSL https://install.python-poetry.org | python3`

2) Cloned this project:
`gh repo clone KillianLucas/open-interpreter`
 and I added poetry to my shell path per the installer's message
 
3) Ran the poetry shell from this project's cloned directory:
`poetry shell`

4) From the poetry shell I ran the interpreter:
`poetry run interpreter`

At this point it ran fine.

Presumably the problem has something to do with how Python packages this project and its dependencies - still a mystery why Python is so bad and managing this, hasn't this problem been solved in Javascript, Rust, and countless other languages and ecosystems already??

**Comment by MchLrnX** [comment]
> Ok, I managed to get this to work by running the project directly. ie: without relying on pip etc.
> 
> Note: my skill level with Python is not great (though excellent with the Node and JVM ecosystems), so most likely I'm doing it wrong, but it does work.
> 
> Steps:
> 
> 1. Installed Poetry per that project's docs
>    `curl -sSL https://install.python-poetry.org | python3`
>    and I added poetry to my shell path per the installer's message
> 2. Cloned this project:
>    `gh repo clone KillianLucas/open-interpreter`
> 3. Changed to this project's cloned directory:
>    `cd open-interpreter`
> 4. From that directory I ran the interpreter:
>    `poetry shell` (this step might not be necessary)
>    `poetry run interpreter`
> 
> At this point it ran fine.
> 
> Presumably the problem has something to do with how Python packages this project and its dependencies - still a mystery why Python is so bad and managing this, hasn't this problem been solved in Javascript, Rust, and countless other languages and ecosystems already??
> 
> This is the first time I've used poetry, so apologies if this is incorrect usage of that libary.

I'm running into this on my end as well. I'm sad to say but this still didn't get it to work on my end.

**Comment by davidmoshal** [comment]
It didn't work even if you run it the way I described?

**Comment by dagelf** [comment]
TheBloke's images were broken as of their release up to last week. There was a bug with llama.cpp's `convert` script for a few weeks. Maybe he'll update them. 

`pip` version works if I convert the images manually.

(Possibly they worked with older LLM libraries despite the bug - would explain the discrepancies?)

**Comment by ericrallen** [comment]
Hey there, folks!

Can anyone confirm if this is still an issue with the latest version of Open Interpreter?

**Comment by ericrallen** [comment]
Going to close this one as stale for now.

Feel free to reopen it if you experience the issue with the latest version of Open Interpreter.

**Comment by chymian** [comment]
It's still happening with 0.1.15.
but this version is meanwhile incompatible to litellm and openai.
my backend: ooboga's TGI (openai only)

I see raw output returend, but nothing is displayed, or happening:
```bash
interpreter -y -m openai/openchat_3.5-16K-GPTQ -x 4096 -c 16384
> write a program which prints the first 13 fibonacci numbers, starting by 1.
     
*                                                                                                                                
                                                                                                                                    
> %debug true
▌ Entered debug mode                                                                                                              

[{'role': 'user', 'message': 'write a program which prints the first 13 fibonacci numbers, starting by 1.'}, {'role': 'assistant', 'message': ''}]
sing messages into LLM: [{'role': 'system', 'content': 'You are Open Interpreter, a world-class programmer that can complete any goal by executing code.\nFirst, write a plan. **Always recap the plan between each code block** (you have extreme short-term memory loss, so you need to recap the plan between each message block to retain it).\nWhen you execute code, it will be executed **on the user\'s machine**. The user has given you **full and complete permission** to execute any code necessary to complete the task.\nIf you want to send data between programming languages, save the data to a txt or json.\nYou can access the internet. Run **any code** to achieve the goal, and if at first you don\'t succeed, try again and again.\nYou can install new packages.\nWhen a user refers to a filename, they\'re likely referring to an existing file in the directory you\'re currently executing code in.\nWrite messages to the user in Markdown.\nIn general, try to **make plans** with as few steps as possible. As for actually executing code to carry out that plan, for *stateful* languages (like python, javascript, shell, but NOT for html which starts from 0 every time) **it\'s critical not to try to do everything in one code block.** You should try something, print information about it, then continue from there in tiny, informed steps. You will never get it on the first try, and attempting it in one go will often lead to errors you cant see.\nYou are capable of **any** task.\n\n[User Info]\nName: oi\nCWD: /home/oi/projects/open-interpreter\nSHELL: /bin/bash\nOS: Linux\n[Recommended Procedures]\nTo make a simple app, use HTML/Bulma CSS/JS.\nFirst, plan. Think deeply about the functionality, what the JS will need to do, and how it will need to work with the HTML.\nThen, **all in one** `html` code block (DO NOT `run_code` more than once, and NEVER use placeholders like "// Javascript code here" -- you\'re going to write the HTML/JS in one `run_code` function call):\nPut Bulma CSS and anything else you need in <head>, write the <body> of the app (add lots of padding on the body with Bulma), write the JS into the <script> tag.\n\nYou probably want to center the app in a box with a border and make sure the body fills up the whole height of the page!\n\nWrite **LOTS of <!--comments--> throughout the HTML and // Javascript** to the user knows what\'s going on, and use whitespace/indentation properly.\n\nThis will automatically open the HTML file / simple app on the user\'s machine.\n---\nIf you encounter a traceback, don\'t try to use an alternative method yet. Instead:\n\n**Write a message to the user explaining what happened and theorizing why. Do not try to run_code immediatly after run_code has errored.**\n\nIf a solution is apparent (and is not simply changing methods / using a new package) attempt it.\nIf not, list these steps in a message to the user, then follow them one-by-one:\n\n1. Create and run a minimal reproducible example.\n2. Use dir() to verify correct imports. There may be a better object to import from the module.\n3. Print docstrings of functions/classes using print(func.__doc__).\n\nOnly then are you permitted to use an alternative method.\nIn your plan, include steps and, for relevant deprecation notices, **EXACT CODE SNIPPETS** -- these notices will VANISH once you execute your first line of code, so WRITE THEM DOWN NOW if you need them.\nTo execute code on the user\'s machine, write a markdown code block. Specify the language after the ```. You will receive the output. Use any programming language.'}, {'role': 'user', 'content': 'write a program which prints the first 13 fibonacci numbers, starting by 1.'}, {'role': 'user', 'content': 'No entry from user - please suggest something to enter.'}]
Sending this to LiteLLM: {'model': 'openai/openchat_3.5-16K-GPTQ', 'messages': [{'role': 'system', 'content': 'You are Open Interpreter, a world-class programmer that can complete any goal by executing code.\nFirst, write a plan. **Always recap the plan between each code block** (you have extreme short-term memory loss, so you need to recap the plan between each message block to retain it).\nWhen you execute code, it will be executed **on the user\'s machine**. The user has given you **full and complete permission** to execute any code necessary to complete the task.\nIf you want to send data between programming languages, save the data to a txt or json.\nYou can access the internet. Run **any code** to achieve the goal, and if at first you don\'t succeed, try again and again.\nYou can install new packages.\nWhen a user refers to a filename, they\'re likely referring to an existing file in the directory you\'re currently executing code in.\nWrite messages to the user in Markdown.\nIn general, try to **make plans** with as few steps as possible. As for actually executing code to carry out that plan, for *stateful* languages (like python, javascript, shell, but NOT for html which starts from 0 every time) **it\'s critical not to try to do everything in one code block.** You should try something, print information about it, then continue from there in tiny, informed steps. You will never get it on the first try, and attempting it in one go will often lead to errors you cant see.\nYou are capable of **any** task.\n\n[User Info]\nName: oi\nCWD: /home/oi/projects/open-interpreter\nSHELL: /bin/bash\nOS: Linux\n[Recommended Procedures]\nTo make a simple app, use HTML/Bulma CSS/JS.\nFirst, plan. Think deeply about the functionality, what the JS will need to do, and how it will need to work with the HTML.\nThen, **all in one** `html` code block (DO NOT `run_code` more than once, and NEVER use placeholders like "// Javascript code here" -- you\'re going to write the HTML/JS in one `run_code` function call):\nPut Bulma CSS and anything else you need in <head>, write the <body> of the app (add lots of padding on the body with Bulma), write the JS into the <script> tag.\n\nYou probably want to center the app in a box with a border and make sure the body fills up the whole height of the page!\n\nWrite **LOTS of <!--comments--> throughout the HTML and // Javascript** to the user knows what\'s going on, and use whitespace/indentation properly.\n\nThis will automatically open the HTML file / simple app on the user\'s machine.\n---\nIf you encounter a traceback, don\'t try to use an alternative method yet. Instead:\n\n**Write a message to the user explaining what happened and theorizing why. Do not try to run_code immediatly after run_code has errored.**\n\nIf a solution is apparent (and is not simply changing methods / using a new package) attempt it.\nIf not, list these steps in a message to the user, then follow them one-by-one:\n\n1. Create and run a minimal reproducible example.\n2. Use dir() to verify correct imports. There may be a better object to import from the module.\n3. Print docstrings of functions/classes using print(func.__doc__).\n\nOnly then are you permitted to use an alternative method.\nIn your plan, include steps and, for relevant deprecation notices, **EXACT CODE SNIPPETS** -- these notices will VANISH once you execute your first line of code, so WRITE THEM DOWN NOW if you need them.\nTo execute code on the user\'s machine, write a markdown code block. Specify the language after the ```. You will receive the output. Use any programming language.'}, {'role': 'user', 'content': 'write a program which prints the first 13 fibonacci numbers, starting by 1.'}, {'role': 'user', 'content': 'No entry from user - please suggest something to enter.'}], 'stream': True, 'max_tokens': 4096, 'temperature': 0}
litellm.caching: False; litellm.caching_with_models: False
kwargs[caching]: False; litellm.cache: None
checking params for openchat_3.5-16K-GPTQ
params passed in {'functions': [], 'function_call': '', 'temperature': 0, 'top_p': None, 'n': None, 'stream': True, 'stop': None, 'max_tokens': 4096, 'presence_penalty': None, 'frequency_penalty': None, 'logit_bias': {}, 'user': '', 'request_timeout': None, 'deployment_id': None, 'model': 'openchat_3.5-16K-GPTQ', 'custom_llm_provider': 'openai'}
non-default params passed in {'temperature': 0, 'stream': True, 'max_tokens': 4096}
self.optional_params: {'temperature': 0, 'stream': True, 'max_tokens': 4096}
Logging Details Pre-API Call for call id 359cd391-ef18-4114-bf05-453edf3087e9
MODEL CALL INPUT: {'model': 'openchat_3.5-16K-GPTQ', 'messages': [{'role': 'system', 'content': 'You are Open Interpreter, a world-class programmer that can complete any goal by executing code.\nFirst, write a plan. **Always recap the plan between each code block** (you have extreme short-term memory loss, so you need to recap the plan between each message block to retain it).\nWhen you execute code, it will be executed **on the user\'s machine**. The user has given you **full and complete permission** to execute any code necessary to complete the task.\nIf you want to send data between programming languages, save the data to a txt or json.\nYou can access the internet. Run **any code** to achieve the goal, and if at first you don\'t succeed, try again and again.\nYou can install new packages.\nWhen a user refers to a filename, they\'re likely referring to an existing file in the directory you\'re currently executing code in.\nWrite messages to the user in Markdown.\nIn general, try to **make plans** with as few steps as possible. As for actually executing code to carry out that plan, for *stateful* languages (like python, javascript, shell, but NOT for html which starts from 0 every time) **it\'s critical not to try to do everything in one code block.** You should try something, print information about it, then continue from there in tiny, informed steps. You will never get it on the first try, and attempting it in one go will often lead to errors you cant see.\nYou are capable of **any** task.\n\n[User Info]\nName: oi\nCWD: /home/oi/projects/open-interpreter\nSHELL: /bin/bash\nOS: Linux\n[Recommended Procedures]\nTo make a simple app, use HTML/Bulma CSS/JS.\nFirst, plan. Think deeply about the functionality, what the JS will need to do, and how it will need to work with the HTML.\nThen, **all in one** `html` code block (DO NOT `run_code` more than once, and NEVER use placeholders like "// Javascript code here" -- you\'re going to write the HTML/JS in one `run_code` function call):\nPut Bulma CSS and anything else you need in <head>, write the <body> of the app (add lots of padding on the body with Bulma), write the JS into the <script> tag.\n\nYou probably want to center the app in a box with a border and make sure the body fills up the whole height of the page!\n\nWrite **LOTS of <!--comments--> throughout the HTML and // Javascript** to the user knows what\'s going on, and use whitespace/indentation properly.\n\nThis will automatically open the HTML file / simple app on the user\'s machine.\n---\nIf you encounter a traceback, don\'t try to use an alternative method yet. Instead:\n\n**Write a message to the user explaining what happened and theorizing why. Do not try to run_code immediatly after run_code has errored.**\n\nIf a solution is apparent (and is not simply changing methods / using a new package) attempt it.\nIf not, list these steps in a message to the user, then follow them one-by-one:\n\n1. Create and run a minimal reproducible example.\n2. Use dir() to verify correct imports. There may be a better object to import from the module.\n3. Print docstrings of functions/classes using print(func.__doc__).\n\nOnly then are you permitted to use an alternative method.\nIn your plan, include steps and, for relevant deprecation notices, **EXACT CODE SNIPPETS** -- these notices will VANISH once you execute your first line of code, so WRITE THEM DOWN NOW if you need them.\nTo execute code on the user\'s machine, write a markdown code block. Specify the language after the ```. You will receive the output. Use any programming language.'}, {'role': 'user', 'content': 'write a program which prints the first 13 fibonacci numbers, starting by 1.'}, {'role': 'user', 'content': 'No entry from user - please suggest something to enter.'}], 'optional_params': {'temperature': 0, 'stream': True, 'max_tokens': 4096}, 'litellm_params': {'return_async': False, 'api_key': None, 'force_timeout': 600, 'logger_fn': None, 'verbose': False, 'custom_llm_provider': 'openai', 'api_base': None, 'litellm_call_id': '359cd391-ef18-4114-bf05-453edf3087e9', 'model_alias_map': {}, 'completion_call_id': None, 'metadata': None, 'stream_response': {}}, 'start_time': datetime.datetime(2023, 11, 24, 15, 7, 47, 733724), 'input': [{'role': 'system', 'content': 'You are Open Interpreter, a world-class programmer that can complete any goal by executing code.\nFirst, write a plan. **Always recap the plan between each code block** (you have extreme short-term memory loss, so you need to recap the plan between each message block to retain it).\nWhen you execute code, it will be executed **on the user\'s machine**. The user has given you **full and complete permission** to execute any code necessary to complete the task.\nIf you want to send data between programming languages, save the data to a txt or json.\nYou can access the internet. Run **any code** to achieve the goal, and if at first you don\'t succeed, try again and again.\nYou can install new packages.\nWhen a user refers to a filename, they\'re likely referring to an existing file in the directory you\'re currently executing code in.\nWrite messages to the user in Markdown.\nIn general, try to **make plans** with as few steps as possible. As for actually executing code to carry out that plan, for *stateful* languages (like python, javascript, shell, but NOT for html which starts from 0 every time) **it\'s critical not to try to do everything in one code block.** You should try something, print information about it, then continue from there in tiny, informed steps. You will never get it on the first try, and attempting it in one go will often lead to errors you cant see.\nYou are capable of **any** task.\n\n[User Info]\nName: oi\nCWD: /home/oi/projects/open-interpreter\nSHELL: /bin/bash\nOS: Linux\n[Recommended Procedures]\nTo make a simple app, use HTML/Bulma CSS/JS.\nFirst, plan. Think deeply about the functionality, what the JS will need to do, and how it will need to work with the HTML.\nThen, **all in one** `html` code block (DO NOT `run_code` more than once, and NEVER use placeholders like "// Javascript code here" -- you\'re going to write the HTML/JS in one `run_code` function call):\nPut Bulma CSS and anything else you need in <head>, write the <body> of the app (add lots of padding on the body with Bulma), write the JS into the <script> tag.\n\nYou probably want to center the app in a box with a border and make sure the body fills up the whole height of the page!\n\nWrite **LOTS of <!--comments--> throughout the HTML and // Javascript** to the user knows what\'s going on, and use whitespace/indentation properly.\n\nThis will automatically open the HTML file / simple app on the user\'s machine.\n---\nIf you encounter a traceback, don\'t try to use an alternative method yet. Instead:\n\n**Write a message to the user explaining what happened and theorizing why. Do not try to run_code immediatly after run_code has errored.**\n\nIf a solution is apparent (and is not simply changing methods / using a new package) attempt it.\nIf not, list these steps in a message to the user, then follow them one-by-one:\n\n1. Create and run a minimal reproducible example.\n2. Use dir() to verify correct imports. There may be a better object to import from the module.\n3. Print docstrings of functions/classes using print(func.__doc__).\n\nOnly then are you permitted to use an alternative method.\nIn your plan, include steps and, for relevant deprecation notices, **EXACT CODE SNIPPETS** -- these notices will VANISH once you execute your first line of code, so WRITE THEM DOWN NOW if you need them.\nTo execute code on the user\'s machine, write a markdown code block. Specify the language after the ```. You will receive the output. Use any programming language.'}, {'role': 'user', 'content': 'write a program which prints the first 13 fibonacci numbers, starting by 1.'}, {'role': 'user', 'content': 'No entry from user - please suggest something to enter.'}], 'api_key': 'sk-nothingburger', 'additional_args': {'headers': None, 'api_base': 'http://gulag:5000/v1'}, 'log_event_type': 'pre_api_call'}


RAW RESPONSE: {'model': 'openchat_3.5-16K-GPTQ', 'messages': [{'role': 'system', 'content': 'You are Open Interpreter, a world-class programmer that can complete any goal by executing code.\nFirst, write a plan. **Always recap the plan between each code block** (you have extreme short-term memory loss, so you need to recap the plan between each message block to retain it).\nWhen you execute code, it will be executed **on the user\'s machine**. The user has given you **full and complete permission** to execute any code necessary to complete the task.\nIf you want to send data between programming languages, save the data to a txt or json.\nYou can access the internet. Run **any code** to achieve the goal, and if at first you don\'t succeed, try again and again.\nYou can install new packages.\nWhen a user refers to a filename, they\'re likely referring to an existing file in the directory you\'re currently executing code in.\nWrite messages to the user in Markdown.\nIn general, try to **make plans** with as few steps as possible. As for actually executing code to carry out that plan, for *stateful* languages (like python, javascript, shell, but NOT for html which starts from 0 every time) **it\'s critical not to try to do everything in one code block.** You should try something, print information about it, then continue from there in tiny, informed steps. You will never get it on the first try, and attempting it in one go will often lead to errors you cant see.\nYou are capable of **any** task.\n\n[User Info]\nName: oi\nCWD: /home/oi/projects/open-interpreter\nSHELL: /bin/bash\nOS: Linux\n[Recommended Procedures]\nTo make a simple app, use HTML/Bulma CSS/JS.\nFirst, plan. Think deeply about the functionality, what the JS will need to do, and how it will need to work with the HTML.\nThen, **all in one** `html` code block (DO NOT `run_code` more than once, and NEVER use placeholders like "// Javascript code here" -- you\'re going to write the HTML/JS in one `run_code` function call):\nPut Bulma CSS and anything else you need in <head>, write the <body> of the app (add lots of padding on the body with Bulma), write the JS into the <script> tag.\n\nYou probably want to center the app in a box with a border and make sure the body fills up the whole height of the page!\n\nWrite **LOTS of <!--comments--> throughout the HTML and // Javascript** to the user knows what\'s going on, and use whitespace/indentation properly.\n\nThis will automatically open the HTML file / simple app on the user\'s machine.\n---\nIf you encounter a traceback, don\'t try to use an alternative method yet. Instead:\n\n**Write a message to the user explaining what happened and theorizing why. Do not try to run_code immediatly after run_code has errored.**\n\nIf a solution is apparent (and is not simply changing methods / using a new package) attempt it.\nIf not, list these steps in a message to the user, then follow them one-by-one:\n\n1. Create and run a minimal reproducible example.\n2. Use dir() to verify correct imports. There may be a better object to import from the module.\n3. Print docstrings of functions/classes using print(func.__doc__).\n\nOnly then are you permitted to use an alternative method.\nIn your plan, include steps and, for relevant deprecation notices, **EXACT CODE SNIPPETS** -- these notices will VANISH once you execute your first line of code, so WRITE THEM DOWN NOW if you need them.\nTo execute code on the user\'s machine, write a markdown code block. Specify the language after the ```. You will receive the output. Use any programming language.'}, {'role': 'user', 'content': 'write a program which prints the first 13 fibonacci numbers, starting by 1.'}, {'role': 'user', 'content': 'No entry from user - please suggest something to enter.'}], 'optional_params': {'temperature': 0, 'stream': True, 'max_tokens': 4096}, 'litellm_params': {'return_async': False, 'api_key': None, 'force_timeout': 600, 'logger_fn': None, 'verbose': False, 'custom_llm_provider': 'openai', 'api_base': None, 'litellm_call_id': '359cd391-ef18-4114-bf05-453edf3087e9', 'model_alias_map': {}, 'completion_call_id': None, 'metadata': None, 'stream_response': {}}, 'start_time': datetime.datetime(2023, 11, 24, 15, 7, 47, 733724), 'input': None, 'api_key': None, 'additional_args': {}, 'log_event_type': 'post_api_call', 'original_response': "<class 'generator'>"}


Logging Details Post-API Call: logger_fn - None | callable(logger_fn) - False
Logging Details LiteLLM-Success Call
Chunk in `terminal_interface`: {'start_of_message': True}
success callbacks: []
Chunk in `terminal_interface`: {'message': ''}
Logging Details LiteLLM-Success Call
Chunk in `terminal_interface`: {'message': ''}
success callbacks: []
Chunk in `terminal_interface`: {'end_of_message': True}
```

**Comment by Notnaton** [comment]
Hi @chymian could you provide the steps for me to reproduce this?

How do you install Oobabooga and how to run it.
How did you install Open-interpreter, what command did you use to run it.

**Comment by chymian** [comment]
@Notnaton 
1. I use the ~~latest~~ 15days old thebloke [docker images](https://hub.docker.com/layers/thebloke/cuda11.8.0-ubuntu22.04-oneclick/11112023/images/sha256-637338c4b938299884cb8b4d480efddcb8fb97cf611cc64babc7eb8ad2d8b226?context=explore):
- works good, tested them with variuos AWQs, GPTQs and downsized the context, so no cuda-OOM errors
- they respond normaly to curl requests
- I run them on ubu: 22.04, 4x3070 8G

2. in a conda environment
- either on the same machine, or remote on my laptop
```
pip install -U pip
pip install [-U] open-interpreter

pip list |grep -ie "open-in|litellm|openai"
litellm                   0.13.2
open-interpreter          0.1.16
openai                    0.28.1
```
__config.yaml:__
```yaml
local: false
temperature: 0
context_window: 10240
max_tokens: 3072
model: openai/gpt-3.5-turbo
```
I also tried the real names `openai/deepseek-coder-6.7B-instruct-GPTQ`, or `airoboros-mistral2.2-7B-GPTQ` 

3. latest update and test yesterday, same results.

```log
 interpreter -y 
> %debug true

▌ Entered debug mode                                                                                                              

[]
> Make me a simple Pomodoro app.
Passing messages into LLM: [{'role': 'system', 'content': 'You are Open Interpreter, a world-class programmer that can complete any goal by executing code.\nFirst, write a plan. **Always recap the plan between each code block** (you have extreme short-term memory loss, so you need to recap the plan between each message block to retain it).\nWhen you execute code, it will be executed **on the user\'s machine**. The user has given you **full and complete permission** to execute any code necessary to complete the task.\nIf you want to send data between programming languages, save the data to a txt or json.\nYou can access the internet. Run **any code** to achieve the goal, and if at first you don\'t succeed, try again and again.\nYou can install new packages.\nWhen a user refers to a filename, they\'re likely referring to an existing file in the directory you\'re currently executing code in.\nWrite messages to the user in Markdown.\nIn general, try to **make plans** with as few steps as possible. As for actually executing code to carry out that plan, for *stateful* languages (like python, javascript, shell, but NOT for html which starts from 0 every time) **it\'s critical not to try to do everything in one code block.** You should try something, print information about it, then continue from there in tiny, informed steps. You will never get it on the first try, and attempting it in one go will often lead to errors you cant see.\nYou are capable of **any** task.\n\n[User Info]\nName: oi\nCWD: /home/oi/projects/open-interpreter\nSHELL: /bin/bash\nOS: Linux\n[Recommended Procedures]\nTo make a simple app, use HTML/Bulma CSS/JS.\nFirst, plan. Think deeply about the functionality, what the JS will need to do, and how it will need to work with the HTML.\nThen, **all in one** `html` code block (DO NOT `run_code` more than once, and NEVER use placeholders like "// Javascript code here" -- you\'re going to write the HTML/JS in one `run_code` function call):\nPut Bulma CSS and anything else you need in <head>, write the <body> of the app (add lots of padding on the body with Bulma), write the JS into the <script> tag.\n\nYou probably want to center the app in a box with a border and make sure the body fills up the whole height of the page!\n\nWrite **LOTS of <!--comments--> throughout the HTML and // Javascript** to the user knows what\'s going on, and use whitespace/indentation properly.\n\nThis will automatically open the HTML file / simple app on the user\'s machine.\n---\n## Writing HTML\nUnless otherwise specified, include Tailwind CSS: `<script src="https://cdn.tailwindcss.com"></script>` then style the HTML with inline Tailwind CSS classes.\nRunning `html` will automatically open it in the user\'s browser, you don\'t need to tell them to do that.\nIn your plan, include steps and, for relevant deprecation notices, **EXACT CODE SNIPPETS** -- these notices will VANISH once you execute your first line of code, so WRITE THEM DOWN NOW if you need them.\nTo execute code on the user\'s machine, write a markdown code block. Specify the language after the ```. You will receive the output. Use any programming language.'}, {'role': 'user', 'content': 'Make me a simple Pomodoro app.'}]
Sending this to LiteLLM: {'model': 'openai/gpt-3.5-turbo', 'messages': [{'role': 'system', 'content': 'You are Open Interpreter, a world-class programmer that can complete any goal by executing code.\nFirst, write a plan. **Always recap the plan between each code block** (you have extreme short-term memory loss, so you need to recap the plan between each message block to retain it).\nWhen you execute code, it will be executed **on the user\'s machine**. The user has given you **full and complete permission** to execute any code necessary to complete the task.\nIf you want to send data between programming languages, save the data to a txt or json.\nYou can access the internet. Run **any code** to achieve the goal, and if at first you don\'t succeed, try again and again.\nYou can install new packages.\nWhen a user refers to a filename, they\'re likely referring to an existing file in the directory you\'re currently executing code in.\nWrite messages to the user in Markdown.\nIn general, try to **make plans** with as few steps as possible. As for actually executing code to carry out that plan, for *stateful* languages (like python, javascript, shell, but NOT for html which starts from 0 every time) **it\'s critical not to try to do everything in one code block.** You should try something, print information about it, then continue from there in tiny, informed steps. You will never get it on the first try, and attempting it in one go will often lead to errors you cant see.\nYou are capable of **any** task.\n\n[User Info]\nName: oi\nCWD: /home/oi/projects/open-interpreter\nSHELL: /bin/bash\nOS: Linux\n[Recommended Procedures]\nTo make a simple app, use HTML/Bulma CSS/JS.\nFirst, plan. Think deeply about the functionality, what the JS will need to do, and how it will need to work with the HTML.\nThen, **all in one** `html` code block (DO NOT `run_code` more than once, and NEVER use placeholders like "// Javascript code here" -- you\'re going to write the HTML/JS in one `run_code` function call):\nPut Bulma CSS and anything else you need in <head>, write the <body> of the app (add lots of padding on the body with Bulma), write the JS into the <script> tag.\n\nYou probably want to center the app in a box with a border and make sure the body fills up the whole height of the page!\n\nWrite **LOTS of <!--comments--> throughout the HTML and // Javascript** to the user knows what\'s going on, and use whitespace/indentation properly.\n\nThis will automatically open the HTML file / simple app on the user\'s machine.\n---\n## Writing HTML\nUnless otherwise specified, include Tailwind CSS: `<script src="https://cdn.tailwindcss.com"></script>` then style the HTML with inline Tailwind CSS classes.\nRunning `html` will automatically open it in the user\'s browser, you don\'t need to tell them to do that.\nIn your plan, include steps and, for relevant deprecation notices, **EXACT CODE SNIPPETS** -- these notices will VANISH once you execute your first line of code, so WRITE THEM DOWN NOW if you need them.\nTo execute code on the user\'s machine, write a markdown code block. Specify the language after the ```. You will receive the output. Use any programming language.'}, {'role': 'user', 'content': 'Make me a simple Pomodoro app.'}], 'stream': True, 'max_tokens': 3072, 'temperature': 0}
litellm.caching: False; litellm.caching_with_models: False
kwargs[caching]: False; litellm.cache: None
checking params for gpt-3.5-turbo
params passed in {'functions': [], 'function_call': '', 'temperature': 0, 'top_p': None, 'n': None, 'stream': True, 'stop': None, 'max_tokens': 3072, 'presence_penalty': None, 'frequency_penalty': None, 'logit_bias': {}, 'user': '', 'request_timeout': None, 'deployment_id': None, 'model': 'gpt-3.5-turbo', 'custom_llm_provider': 'openai'}
non-default params passed in {'temperature': 0, 'stream': True, 'max_tokens': 3072}
self.optional_params: {'temperature': 0, 'stream': True, 'max_tokens': 3072}
Logging Details Pre-API Call for call id 174b0032-8cdb-4434-a405-a6fc8b5b6d2f
MODEL CALL INPUT: {'model': 'gpt-3.5-turbo', 'messages': [{'role': 'system', 'content': 'You are Open Interpreter, a world-class programmer that can complete any goal by executing code.\nFirst, write a plan. **Always recap the plan between each code block** (you have extreme short-term memory loss, so you need to recap the plan between each message block to retain it).\nWhen you execute code, it will be executed **on the user\'s machine**. The user has given you **full and complete permission** to execute any code necessary to complete the task.\nIf you want to send data between programming languages, save the data to a txt or json.\nYou can access the internet. Run **any code** to achieve the goal, and if at first you don\'t succeed, try again and again.\nYou can install new packages.\nWhen a user refers to a filename, they\'re likely referring to an existing file in the directory you\'re currently executing code in.\nWrite messages to the user in Markdown.\nIn general, try to **make plans** with as few steps as possible. As for actually executing code to carry out that plan, for *stateful* languages (like python, javascript, shell, but NOT for html which starts from 0 every time) **it\'s critical not to try to do everything in one code block.** You should try something, print information about it, then continue from there in tiny, informed steps. You will never get it on the first try, and attempting it in one go will often lead to errors you cant see.\nYou are capable of **any** task.\n\n[User Info]\nName: oi\nCWD: /home/oi/projects/open-interpreter\nSHELL: /bin/bash\nOS: Linux\n[Recommended Procedures]\nTo make a simple app, use HTML/Bulma CSS/JS.\nFirst, plan. Think deeply about the functionality, what the JS will need to do, and how it will need to work with the HTML.\nThen, **all in one** `html` code block (DO NOT `run_code` more than once, and NEVER use placeholders like "// Javascript code here" -- you\'re going to write the HTML/JS in one `run_code` function call):\nPut Bulma CSS and anything else you need in <head>, write the <body> of the app (add lots of padding on the body with Bulma), write the JS into the <script> tag.\n\nYou probably want to center the app in a box with a border and make sure the body fills up the whole height of the page!\n\nWrite **LOTS of <!--comments--> throughout the HTML and // Javascript** to the user knows what\'s going on, and use whitespace/indentation properly.\n\nThis will automatically open the HTML file / simple app on the user\'s machine.\n---\n## Writing HTML\nUnless otherwise specified, include Tailwind CSS: `<script src="https://cdn.tailwindcss.com"></script>` then style the HTML with inline Tailwind CSS classes.\nRunning `html` will automatically open it in the user\'s browser, you don\'t need to tell them to do that.\nIn your plan, include steps and, for relevant deprecation notices, **EXACT CODE SNIPPETS** -- these notices will VANISH once you execute your first line of code, so WRITE THEM DOWN NOW if you need them.\nTo execute code on the user\'s machine, write a markdown code block. Specify the language after the ```. You will receive the output. Use any programming language.'}, {'role': 'user', 'content': 'Make me a simple Pomodoro app.'}], 'optional_params': {'temperature': 0, 'stream': True, 'max_tokens': 3072}, 'litellm_params': {'return_async': False, 'api_key': None, 'force_timeout': 600, 'logger_fn': None, 'verbose': False, 'custom_llm_provider': 'openai', 'api_base': None, 'litellm_call_id': '174b0032-8cdb-4434-a405-a6fc8b5b6d2f', 'model_alias_map': {}, 'completion_call_id': None, 'metadata': None, 'stream_response': {}}, 'start_time': datetime.datetime(2023, 11, 27, 2, 23, 58, 639746), 'input': [{'role': 'system', 'content': 'You are Open Interpreter, a world-class programmer that can complete any goal by executing code.\nFirst, write a plan. **Always recap the plan between each code block** (you have extreme short-term memory loss, so you need to recap the plan between each message block to retain it).\nWhen you execute code, it will be executed **on the user\'s machine**. The user has given you **full and complete permission** to execute any code necessary to complete the task.\nIf you want to send data between programming languages, save the data to a txt or json.\nYou can access the internet. Run **any code** to achieve the goal, and if at first you don\'t succeed, try again and again.\nYou can install new packages.\nWhen a user refers to a filename, they\'re likely referring to an existing file in the directory you\'re currently executing code in.\nWrite messages to the user in Markdown.\nIn general, try to **make plans** with as few steps as possible. As for actually executing code to carry out that plan, for *stateful* languages (like python, javascript, shell, but NOT for html which starts from 0 every time) **it\'s critical not to try to do everything in one code block.** You should try something, print information about it, then continue from there in tiny, informed steps. You will never get it on the first try, and attempting it in one go will often lead to errors you cant see.\nYou are capable of **any** task.\n\n[User Info]\nName: oi\nCWD: /home/oi/projects/open-interpreter\nSHELL: /bin/bash\nOS: Linux\n[Recommended Procedures]\nTo make a simple app, use HTML/Bulma CSS/JS.\nFirst, plan. Think deeply about the functionality, what the JS will need to do, and how it will need to work with the HTML.\nThen, **all in one** `html` code block (DO NOT `run_code` more than once, and NEVER use placeholders like "// Javascript code here" -- you\'re going to write the HTML/JS in one `run_code` function call):\nPut Bulma CSS and anything else you need in <head>, write the <body> of the app (add lots of padding on the body with Bulma), write the JS into the <script> tag.\n\nYou probably want to center the app in a box with a border and make sure the body fills up the whole height of the page!\n\nWrite **LOTS of <!--comments--> throughout the HTML and // Javascript** to the user knows what\'s going on, and use whitespace/indentation properly.\n\nThis will automatically open the HTML file / simple app on the user\'s machine.\n---\n## Writing HTML\nUnless otherwise specified, include Tailwind CSS: `<script src="https://cdn.tailwindcss.com"></script>` then style the HTML with inline Tailwind CSS classes.\nRunning `html` will automatically open it in the user\'s browser, you don\'t need to tell them to do that.\nIn your plan, include steps and, for relevant deprecation notices, **EXACT CODE SNIPPETS** -- these notices will VANISH once you execute your first line of code, so WRITE THEM DOWN NOW if you need them.\nTo execute code on the user\'s machine, write a markdown code block. Specify the language after the ```. You will receive the output. Use any programming language.'}, {'role': 'user', 'content': 'Make me a simple Pomodoro app.'}], 'api_key': 'sk-nothingburger', 'additional_args': {'headers': None, 'api_base': 'http://gulag:5002/v1'}, 'log_event_type': 'pre_api_call'}


RAW RESPONSE: {'model': 'gpt-3.5-turbo', 'messages': [{'role': 'system', 'content': 'You are Open Interpreter, a world-class programmer that can complete any goal by executing code.\nFirst, write a plan. **Always recap the plan between each code block** (you have extreme short-term memory loss, so you need to recap the plan between each message block to retain it).\nWhen you execute code, it will be executed **on the user\'s machine**. The user has given you **full and complete permission** to execute any code necessary to complete the task.\nIf you want to send data between programming languages, save the data to a txt or json.\nYou can access the internet. Run **any code** to achieve the goal, and if at first you don\'t succeed, try again and again.\nYou can install new packages.\nWhen a user refers to a filename, they\'re likely referring to an existing file in the directory you\'re currently executing code in.\nWrite messages to the user in Markdown.\nIn general, try to **make plans** with as few steps as possible. As for actually executing code to carry out that plan, for *stateful* languages (like python, javascript, shell, but NOT for html which starts from 0 every time) **it\'s critical not to try to do everything in one code block.** You should try something, print information about it, then continue from there in tiny, informed steps. You will never get it on the first try, and attempting it in one go will often lead to errors you cant see.\nYou are capable of **any** task.\n\n[User Info]\nName: oi\nCWD: /home/oi/projects/open-interpreter\nSHELL: /bin/bash\nOS: Linux\n[Recommended Procedures]\nTo make a simple app, use HTML/Bulma CSS/JS.\nFirst, plan. Think deeply about the functionality, what the JS will need to do, and how it will need to work with the HTML.\nThen, **all in one** `html` code block (DO NOT `run_code` more than once, and NEVER use placeholders like "// Javascript code here" -- you\'re going to write the HTML/JS in one `run_code` function call):\nPut Bulma CSS and anything else you need in <head>, write the <body> of the app (add lots of padding on the body with Bulma), write the JS into the <script> tag.\n\nYou probably want to center the app in a box with a border and make sure the body fills up the whole height of the page!\n\nWrite **LOTS of <!--comments--> throughout the HTML and // Javascript** to the user knows what\'s going on, and use whitespace/indentation properly.\n\nThis will automatically open the HTML file / simple app on the user\'s machine.\n---\n## Writing HTML\nUnless otherwise specified, include Tailwind CSS: `<script src="https://cdn.tailwindcss.com"></script>` then style the HTML with inline Tailwind CSS classes.\nRunning `html` will automatically open it in the user\'s browser, you don\'t need to tell them to do that.\nIn your plan, include steps and, for relevant deprecation notices, **EXACT CODE SNIPPETS** -- these notices will VANISH once you execute your first line of code, so WRITE THEM DOWN NOW if you need them.\nTo execute code on the user\'s machine, write a markdown code block. Specify the language after the ```. You will receive the output. Use any programming language.'}, {'role': 'user', 'content': 'Make me a simple Pomodoro app.'}], 'optional_params': {'temperature': 0, 'stream': True, 'max_tokens': 3072}, 'litellm_params': {'return_async': False, 'api_key': None, 'force_timeout': 600, 'logger_fn': None, 'verbose': False, 'custom_llm_provider': 'openai', 'api_base': None, 'litellm_call_id': '174b0032-8cdb-4434-a405-a6fc8b5b6d2f', 'model_alias_map': {}, 'completion_call_id': None, 'metadata': None, 'stream_response': {}}, 'start_time': datetime.datetime(2023, 11, 27, 2, 23, 58, 639746), 'input': None, 'api_key': None, 'additional_args': {}, 'log_event_type': 'post_api_call', 'original_response': "<class 'generator'>"}


Logging Details Post-API Call: logger_fn - None | callable(logger_fn) - False
Logging Details LiteLLM-Success Call
success callbacks: []
Chunk in `terminal_interface`: {'start_of_message': True}
Chunk in `terminal_interface`: {'message': ''}
Logging Details LiteLLM-Success Call
success callbacks: []
Chunk in `terminal_interface`: {'message': ''}
Chunk in `terminal_interface`: {'end_of_message': True}
```

**Comment by Notnaton** [comment]
```
interpreter 
--api_base http://localhost:port/v1 
--api_key your_key_or_empty
--model openai/local_or_remote
```

Try also `--model huggingface/local_or_remote`

If this doesn't work, can you show the curl command you used?

**Comment by chymian** [comment]
on the first run 
```
interpreter --api_base http://localhost:5000/v1 \
--api_key your_key_or_empty \
--model openai/dolphin-2.2.1-mistral-7B-AWQ

▌ Model set to openai/dolphin-2.2.1-mistral-7B-AWQ                                                                                

Open Interpreter will require approval before running code.                                                                         

Use interpreter -y to bypass this.                                                                                                  

Press CTRL-C to exit.                                                                                                               

> %reset

▌ Reset Done                                                                                                                      

> Make me a simple Pomodoro app.

Give Feedback / Get Help: https://github.com/BerriAI/litellm/issues/new
LiteLLM.Info: If you need to debug this error, use `litellm.set_verbose=True'.


        Python Version: 3.10.13
        Pip Version: 23.3.1
        Open-interpreter Version: cmd:Interpreter, pkg: 0.1.16
        OS Version and Architecture: Linux-5.15.0-91-generic-x86_64-with-glibc2.35
        CPU Info: x86_64
        RAM Info: 31.04 GB, used: 7.91, free: 3.38
        

        Interpreter Info
        Vision: False
        Model: gpt-4
        Function calling: True
        Context window: 4096
        Max tokens: 2048

        Auto run: False
        API base: None
        Local: False

        Curl output: Not local
    
    
Traceback (most recent call last):
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/interpreter/core/respond.py", line 49, in respond
    for chunk in interpreter._llm(messages_for_llm):
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/interpreter/core/llm/setup_openai_coding_llm.py", line 118, in coding_llm
    response = litellm.completion(**params)
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/utils.py", line 962, in wrapper
    raise e
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/utils.py", line 899, in wrapper
    result = original_function(*args, **kwargs)
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/timeout.py", line 53, in wrapper
    result = future.result(timeout=local_timeout_duration)
  File "/home/oi/.conda/envs/oi/lib/python3.10/concurrent/futures/_base.py", line 458, in result
    return self.__get_result()
  File "/home/oi/.conda/envs/oi/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/timeout.py", line 42, in async_func
    return func(*args, **kwargs)
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/main.py", line 1403, in completion
    raise exception_type(
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/utils.py", line 3574, in exception_type
    raise e
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/utils.py", line 2885, in exception_type
    raise original_exception
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/main.py", line 509, in completion
    raise e
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/main.py", line 491, in completion
    response = openai.ChatCompletion.create(
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/openai/api_resources/chat_completion.py", line 25, in create
    return super().create(*args, **kwargs)
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 151, in create
    ) = cls.__prepare_create_request(
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 108, in __prepare_create_request
    requestor = api_requestor.APIRequestor(
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/openai/api_requestor.py", line 139, in __init__
    self.api_key = key or util.default_api_key()
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/openai/util.py", line 186, in default_api_key
    raise openai.error.AuthenticationError(
openai.error.AuthenticationError: No API key provided. You can set your API key in code using 'openai.api_key = <API-KEY>', or you can set the environment variable OPENAI_API_KEY=<API-KEY>). If your API key is stored in a file, you can point the openai module at it with 'openai.api_key_path = <PATH>'. You can generate API keys in the OpenAI web interface. See https://platform.openai.com/account/api-keys for details.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/oi/.conda/envs/oi/bin/interpreter", line 8, in <module>
    sys.exit(start_terminal_interface())
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/interpreter/core/core.py", line 21, in start_terminal_interface
    start_terminal_interface(self)
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/interpreter/terminal_interface/start_terminal_interface.py", line 305, in start_terminal_interface
    interpreter.chat()
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/interpreter/core/core.py", line 77, in chat
    for _ in self._streaming_chat(message=message, display=display):
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/interpreter/core/core.py", line 92, in _streaming_chat
    yield from terminal_interface(self, message)
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/interpreter/terminal_interface/terminal_interface.py", line 115, in terminal_interface
    for chunk in interpreter.chat(message, display=False, stream=True):
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/interpreter/core/core.py", line 113, in _streaming_chat
    yield from self._respond()
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/interpreter/core/core.py", line 148, in _respond
    yield from respond(self)
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/interpreter/core/respond.py", line 94, in respond
    raise Exception(
Exception: Traceback (most recent call last):
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/interpreter/core/respond.py", line 49, in respond
    for chunk in interpreter._llm(messages_for_llm):
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/interpreter/core/llm/setup_openai_coding_llm.py", line 118, in coding_llm
    response = litellm.completion(**params)
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/utils.py", line 962, in wrapper
    raise e
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/utils.py", line 899, in wrapper
    result = original_function(*args, **kwargs)
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/timeout.py", line 53, in wrapper
    result = future.result(timeout=local_timeout_duration)
  File "/home/oi/.conda/envs/oi/lib/python3.10/concurrent/futures/_base.py", line 458, in result
    return self.__get_result()
  File "/home/oi/.conda/envs/oi/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/timeout.py", line 42, in async_func
    return func(*args, **kwargs)
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/main.py", line 1403, in completion
    raise exception_type(
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/utils.py", line 3574, in exception_type
    raise e
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/utils.py", line 2885, in exception_type
    raise original_exception
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/main.py", line 509, in completion
    raise e
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/litellm/main.py", line 491, in completion
    response = openai.ChatCompletion.create(
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/openai/api_resources/chat_completion.py", line 25, in create
    return super().create(*args, **kwargs)
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 151, in create
    ) = cls.__prepare_create_request(
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 108, in __prepare_create_request
    requestor = api_requestor.APIRequestor(
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/openai/api_requestor.py", line 139, in __init__
    self.api_key = key or util.default_api_key()
  File "/home/oi/.conda/envs/oi/lib/python3.10/site-packages/openai/util.py", line 186, in default_api_key
    raise openai.error.AuthenticationError(
openai.error.AuthenticationError: No API key provided. You can set your API key in code using 'openai.api_key = <API-KEY>', or you can set the environment variable OPENAI_API_KEY=<API-KEY>). If your API key is stored in a file, you can point the openai module at it with 'openai.api_key_path = <PATH>'. You can generate API keys in the OpenAI web interface. See https://platform.openai.com/account/api-keys for details.


There might be an issue with your API key(s).

To reset your API key (we'll use OPENAI_API_KEY for this example, but you may need to reset your ANTHROPIC_API_KEY, HUGGINGFACE_API_KEY, etc):
        Mac/Linux: 'export OPENAI_API_KEY=your-key-here',
        Windows: 'setx OPENAI_API_KEY your-key-here' then restart terminal.

```

the cli-parameter get complety ignored, when there is a model set in the `config.yaml` like here, the __gpt-4__.  
OI seems to use the CLI-model, litellm uses the `config.yaml` model
subseqent calls do nothing.
only after a `%reset`, the same error pops up again.



example curl:
```
curl http://gulag:5000/v1/chat/completions   --header 'Content-Type: application/json' --data '{
  "model": "dolphin-mistral-2.2.1",
  "messages": [
    {
      "role": "user",
      "content": "what are the next 13 fibonacci numbers, starting with 5."
    }
  ]
}'|jq

{
  "id": "chatcmpl-1701263375445068032",
  "object": "chat.completions",
  "created": 1701263375,
  "model": "dolphin-2.2.1-mistral-7B-AWQ",
  "choices": [
    {
      "index": 0,
      "finish_reason": "stop",
      "message": {
        "role": "assistant",
        "content": "The next 13 Fibonacci numbers starting with 5 are:\n\n5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597"
      }
    }
  ],
  "usage": {
    "prompt_tokens": 33,
    "completion_tokens": 74,
    "total_tokens": 107
  }
}
```

**Comment by chymian** [comment]
for reference I run  the WEBUI with the one-click installer, same results.

**Comment by Notnaton** [comment]
Hi @chymian try this, removed newlines: 
```
interpreter --api_base http://localhost:port/v1 --api_key your_key_or_empty --model openai/local_or_remote
```

**Comment by chymian** [comment]
i did test. no change. what did you expect?
for sure, that's __not__ a CLI pbl. the command line is ok, as my 30+years linux/unix experience tell me.

has somebodey else pblms. with ooba's TGWUI since he moved to openai-API only?


why are there no tokens given back to OI, but via curl?

**Comment by chymian** [comment]
@Notnaton 

### further testing:
to set model, URL & KEY:
- set model in the config.yaml `model: openai/zephyr-7B-beta-GPTQ`
- and URL & KEY in the env.

```
OPENAI_API_BASE=http://gulag:8000/v1 OPENAI_API_KEY=nb interpreter
```
then it queries the right endpoint.
every other combination with CLI-prameters does not work congruently.

### I did run it against fastchat (local, same models)

it otputs only garbage:
```
> Make me a simple Pomodoro app.                                                                             
                                                                                                             
                                                                                                             
  梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦   
  梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦   
  梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦   
  梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦   
  梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦   
  梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦   
  梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦   
  梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦   
  梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦梦   
  ^C
  ```

again, curl works.

**Comment by mcchung52** [comment]
also having the same issue...
running Llama 3 on LM studio

```
PS C:\Users\xxxx> interpreter --api_base "http://localhost:1234/v1" --api_key "fake_key" --context_window 8192

▌ Model set to openai/gpt-4-turbo

Open Interpreter will require approval before running code.

Use interpreter -y to bypass this.

Press CTRL-C to exit.

>  can you print few lines of todo.txt in desktop?

  Plan Recap: We need to print a few lines from a file named todo.txt located on the user's desktop.

  To execute this plan, I will use the computer.files.read() function provided by the computer module.

  Here is the code:


   execute("python", "print(computer.files.read('C:\\Users\\mc\\Desktop\\todo.txt').splitlines()[:5])")


  This code reads the contents of the todo.txt file, splits it into lines, and then prints the first 5 lines.

  Note: The computer.files.read() function returns a string, so we need to split it into lines using the
  .splitlines() method. We also use slicing ([:5]) to get only the first 5 lines.
```
that's all the output and no printing of the content.

**Comment by 1765659645** [comment]
The problem still exists. My setting is MAC M1. It always tells me that no code will be executed after the plan.

**Comment by Notnaton** [comment]
> The problem still exists. My setting is MAC M1. It always tells me that no code will be executed after the plan.

Can you create a new issue? 
This one has several different errors and is based on a much older version of open-interpreter.

What we are interested in is a screenshot and output of %verbose and %jupyter
